lib/trivial.nix

```
@@ -13,6 +13,19 @@ rec {
     # The value to return
     x: x;
 
+  /* imports a flake.nix without acknowledging its lock file, useful for
+     referencing subflakes from a parent flake. The second argument allows
+     specifying the inputs of this flake.
+
+     Example:
+       callLocklessFlake ./directoryContainingFlake { inherit nixpkgs; }
+  */
+  callLocklessFlake = path: inputs: let
```

```suggestion
  callLocklessFlake = { path, inputs ? {} }: let
```

Make this extensible. For instance, might want to add a `sourceInfo` in the future; who knows what else.

---

this should be almost exactly the api proposed here: https://github.com/NixOS/nix/issues/6228